### PR TITLE
[extension/httpforwarder] remove httpforwarder package

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Remove extension named httpforwarder, use httpforwarderextension instead.

**Link to tracking Issue:**
#24171

### CODE REVIEW

**Review by andrzej-stencel — APPROVED** [review]
:ok_hand:

File: .chloggen/remove_httpforwarder.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: [api, user]
```
Comment (dmitryax) [inline .chloggen/remove_httpforwarder.yaml:27]: Why `user`? I'd expect it to be a no-op change for end users

Comment (astencel-sumo) [reply .chloggen/remove_httpforwarder.yaml:27]: > Why `user`? I'd expect it to be a no-op change for end users

That's a good point. The configuration does not change. Unless we treat distro builders as users? they need to update their ocb manifests, right?

Comment (atoulme) [reply .chloggen/remove_httpforwarder.yaml:27]: right, what Andrzej said is probably what gave me a bit of thinking in adding to the user changelog. I don't have a strong opinion about it, I probably thought more is better than missing to mention it.

Comment (dmitryax) [reply .chloggen/remove_httpforwarder.yaml:27]: Ok, sounds good

**Pull request merged**